redis/src/cluster_async/mod.rs

```diff
@@ -195,18 +304,116 @@ type ConnectionMap<C> = connections_container::ConnectionsMap<ConnectionFuture<C
-struct ClusterConnInner<C> {
-    inner: Core<C>,
+    // return address of node for slot
+    pub(crate) async fn get_address_from_slot(
+        &self,
+        slot: u16,
+        slot_addr: SlotAddr,
+    ) -> Option<String> {
+        self.conn_lock
+            .read()
+            .await
+            .slot_map
+            .get_node_address_for_slot(slot, slot_addr)
+    }
+
+    // return epoch of node
+    pub(crate) async fn get_address_epoch(&self, node_address: &str) -> Result<u64, RedisError> {
+        let command = cmd("CLUSTER").arg("INFO").to_owned();
+        let node_conn = self
+            .conn_lock
+            .read()
+            .await
+            .connection_for_address(node_address);
+        match node_conn {
+            Some((_, conn)) => {
+                let cluster_info = conn.await.req_packed_command(&command).await;
+                match cluster_info {
+                    Ok(value) => {
+                        let info_dict: Result<InfoDict, RedisError> =
+                            FromRedisValue::from_redis_value(&value);
+                        if let Ok(info_dict) = info_dict {
+                            let epoch = info_dict.get("cluster_my_epoch");
+                            if let Some(epoch) = epoch {
+                                Ok(epoch)
+                            } else {
+                                Err(RedisError::from((
+                                    ErrorKind::ResponseError,
+                                    "Failed to get epoch from cluster info",
+                                )))
+                            }
+                        } else {
+                            Err(RedisError::from((
+                                ErrorKind::ResponseError,
+                                "Failed to parse cluster info",
+                            )))
+                        }
+                    }
+                    Err(redis_error) => Err(redis_error),
+                }
+            }
+            None => Err(RedisError::from((
+                ErrorKind::IoError,
+                "Failed to get connection for address",
+            ))),
+        }
+    }
+
+    // return slots of node
+    pub(crate) async fn get_slots_of_address(&self, node_address: &str) -> Vec<u16> {
+        self.conn_lock
+            .read()
+            .await
+            .slot_map
+            .get_slots_of_node(node_address)
+    }
+
+    // Route command to the given address
+    pub(crate) async fn route_command_inner(
+        core: Arc<InnerCore<C>>,
+        cmd: Cmd,
+        address: &str,
+    ) -> RedisResult<Value> {
+        let node_conn = ClusterConnInner::get_connection(
+            InternalSingleNodeRouting::ByAddress(address.to_string()),
+            core,
+        )
+        .await;
+        match node_conn {
+            Ok((_, mut conn)) => {
+                let command_response = conn.req_packed_command(&cmd).await;
+                match command_response {
```

Comment:
seems redundant to me (the "match" block does nothing, so you might as well return the result directly)
Changing lines 404 -> 408 to this have the exact effect:

```rust
conn.req_packed_command(&cmd).await
```